sklearn/metrics/tests/test_common.py

```diff
@@ -1116,3 +1116,18 @@ def test_no_averaging_labels():
             score_labels = metric(y_true, y_pred, labels=labels, average=None)
             score = metric(y_true, y_pred, average=None)
             assert_array_equal(score_labels, score[inverse_labels])
+
+
+@ignore_warnings
+def test_all_true_pos_label():
+    # Correct predictions should still work when there are only 1s in the
+    # actual set
+    examples = np.array([0, 1, 1, 0, 1, 1])
+    all_ones = np.array([1, 1, 1])
+
+    for name in METRICS_WITH_POS_LABEL:
+        if not name == 'roc_curve':
+            metric = ALL_METRICS[name]
+            perfect_score = metric(examples, examples, pos_label=1)
```

Comment:
metric(examples, examples) is not necessarily perfect prediction score if the second argument is meant to be a score rather than a label. You've already seen that in ROC AUC. Is there some better way to characterise metrics for which 0 score does not necessarily mean "predicting the negative class", so that we can give the excepted metrics some name (like how THRESHOLDED_METRICS is used elsewhere in this file)?

Reply:
Having a tough time thinking of a snappy name, but would something like `METRICS_TARGET_VS_PREDICTION_WITH_POS_LABEL` work? It would then encompass all the functions in `METRICS_WITH_POS_LABEL` except for `roc_curve`.

Reply:
`POSITIVE_SCORE_MEANS_POSITIVE_CLASS`?